src/app/shared/post-it-collections/collection-name/post-it-collection-name.component.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {PostItCollectionModel} from '../post-it-collection-model';
+import {CollectionModel} from '../../../core/store/collections/collection.model';
+import {LumeerError} from '../../../core/error/lumeer.error';
+
+@Component({
+  selector: 'post-it-collection-name',
+  templateUrl: './post-it-collection-name.component.html',
+  styleUrls: ['./post-it-collection-name.component.scss']
+})
+export class PostItCollectionNameComponent {
+
+  private readonly COLLECTION_NAME_COLLAPSED_HEIGHT = 42;
+
+  private readonly COLLECTION_NAME_MAX_HEIGHT = 500;
+
+  private readonly COLLECTION_PLACEHOLDER_NAME = 'Name';
+
+  private readonly COLLECTION_WHITESPACE_COLLAPSED = 'nowrap';
+
+  private readonly COLLECTION_WHITESPACE_EXPANDED = 'normal';
+
+  private _postIt: PostItCollectionModel;
+
+  @Input()
+  set postIt(value: PostItCollectionModel) {
+    this._postIt = value;
+    this.extractCollectionName(value.collection);
```

Comment:
Why do you complicate this component if you do not really need the whole post-it object here? You can just have an input with the collection name here and copy its value to the input in `ngOnChanges` method.

Reply:
Because I use the postIt attributes in the component.

Reply:
You don't really need any of its attributes in this component. Try not to over-complicate things and always think about the simplest solution (with the least amount of code).